var/spack/repos/builtin/packages/arm/package.py

```diff
@@ -68,20 +134,30 @@ def cc(self):
         assert self.spec.concrete, msg
         if self.spec.external:
             return self.spec.extra_attributes['compilers'].get('c', None)
-        return str(self.spec.prefix.bin.armclang)
+        return join_path(get_acfl_prefix(self.spec), 'bin', 'armclang')
 
     @property
     def cxx(self):
         msg = "cannot retrieve C++ compiler [spec is not concrete]"
         assert self.spec.concrete, msg
         if self.spec.external:
             return self.spec.extra_attributes['compilers'].get('cxx', None)
-        return os.path.join(self.spec.prefix.bin, 'armclang++')
+        return join_path(get_acfl_prefix(self.spec), 'bin', 'armclang++')
 
     @property
     def fortran(self):
         msg = "cannot retrieve Fortran compiler [spec is not concrete]"
         assert self.spec.concrete, msg
         if self.spec.external:
             return self.spec.extra_attributes['compilers'].get('fortran', None)
-        return str(self.spec.prefix.bin.armflang)
+        return join_path(get_acfl_prefix(self.spec), 'bin', 'armflang')
+
+    def setup_run_environment(self, env):
+        arm_dir = get_acfl_prefix(self.spec)
+        env.set("ARM_LINUX_COMPILER_DIR", arm_dir)
+        env.set("ARM_LINUX_COMPILER_INCLUDES", join_path(arm_dir, 'includes'))
+        env.prepend_path("LD_LIBRARY_PATH", join_path(arm_dir, 'lib'))
+        env.prepend_path("PATH", join_path(arm_dir, 'bin'))
+        env.prepend_path("CPATH", join_path(arm_dir, 'include'))
```

Comment:
Are these three variables needed here? Usually we tend not to add them in setup run environment, so I was wondering what is the use case.